src/libcore/hint.rs

```diff
@@ -118,13 +118,6 @@ pub fn black_box<T>(dummy: T) -> T {
     // this. LLVM's intepretation of inline assembly is that it's, well, a black
     // box. This isn't the greatest implementation since it probably deoptimizes
     // more than we want, but it's so far good enough.
-    #[cfg(not(any(
-        target_arch = "asmjs",
-        all(
-            target_arch = "wasm32",
-            target_os = "emscripten"
-        )
-    )))]
```

Comment:
Since we use the upstream LLVM backend now, we can handle inline assembly just fine.

Reply:
Could the `read_volatile` fallback implementation below be removed since it's no longer necessary?

Reply:
Is it possible for out-of-tree targets to disable `asm!` support? I guess if it were, this code wouldn't compile for those targets. Since no other target prohibits `asm!`, do you think it would make sense to remove that target option entirely? Another complication is that although we support inline assembly, I believe it is still buggy and should probably not be used realistically yet.

Reply:
Oh sorry forgot to repond to this, but afaik you can't disable `asm!` with target configs, but the diff looks good now!